src/history/hs_conn.c

```diff
@@ -133,22 +133,35 @@ int
 __wt_hs_open(WT_SESSION_IMPL *session, const char **cfg)
 {
     WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *hs_session;
 
     conn = S2C(session);
+    hs_session = NULL;
 
     /* Read-only and in-memory configurations don't need the history store table. */
     if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /*
+     * A new session is necessary to open the HS file rather than the default session. Connection
+     * API calls have the ability to change the default session for connections at any stage of
+     * recovery.
```

Comment:
May be "It is necessary to create a new session to initialize the HS file because the default session can be used by other tasks concurrently during recovery."

Reply:
Fixed as suggested.